<launch>
    <!-- INPUT ARGUMENTS -->
    <!-- If to use the robot or not (useful for debugging) -->
    <arg name="use_robot" default="true"/>
    <!-- If the system should internally recover from error -->
    <arg name="internal_recovery" default="true"/>
    <!-- If to use the speech to text or not -->
    <arg name="use_tts"   default="true"/>
    <!-- If to use the text to speech or not -->
    <arg name="use_stt"   default="true"/>
    <!-- print level (shared throughout the control code) -->
    <arg name="print_level" default="2"/>

    <!-- ARUCO -->
    <include file="$(find human_robot_collaboration)/launch/baxter_aruco.launch">
        <arg name="arm" value= "left" />
    </include>
    <include file="$(find human_robot_collaboration)/launch/baxter_aruco.launch">
        <arg name="arm" value="right" />
    </include>
    <include file="$(find human_robot_collaboration)/launch/baxter_display.launch">
        <arg name="onscreen_duration" value = "5.0" />
    </include>

    <!-- ROSBRIDGE -->
    <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"/>

    <!-- Objects database for the left arm -->
    <rosparam param = "action_provider/objects_left">
        "seat"       : 198
        "chair_back" : 201
        "leg_1"      : 150
        "leg_2"      : 151
        "leg_3"      : 152
        "leg_4"      : 153
        "leg_5"      : 154
        "leg_6"      : 155
        "leg_7"      : 156
    </rosparam>

    <!-- Objects database for the right arm -->
    <rosparam param = "action_provider/objects_right">
        "foot_1"        : 10
        "foot_2"        : 11
        "foot_3"        : 12
        "foot_4"        : 13
        "front_1"       : 14
        "front_2"       : 15
        "top_1"         : 16
        "top_2"         : 17
        "back_1"        : 18
        "back_2"        : 19
        "screwdriver_1" : 20
        "front_3"       : 22
        "front_4"       : 23
    </rosparam>

    <!-- CONTROL -->
    <param name="action_provider/use_robot" value="$(arg use_robot)"/>
    <param name="action_provider/internal_recovery" value="$(arg internal_recovery)"/>
    <param name="print_level" value="$(arg print_level)" />

    <node pkg="human_robot_collaboration" type="sp_pred_action_provider" name="sp_pred_action_provider" output="screen">
        <remap from="/markers/left"  to="/baxter_aruco_left/markers"/>
        <remap from="/markers/right" to="/baxter_aruco_right/markers"/>
    </node>

    <!-- Text to Speech -->
    <group if="$(arg use_tts)">
        <include file="$(find svox_tts)/launch/svox_tts.launch" />
    </group>

    <!-- Speech to Text -->
    <group if="$(arg use_stt)">
        <include file="$(find ros_speech2text)/launch/ros_speech2text.launch" />
    </group>

</launch>
